### PR TITLE
Replace TSLint with ESLint

### PR DESCRIPTION
**Pull request by Half-Shot**
Most of our bridges have been moved to eslint, but this one got left behind. This PR corrects that injustice. The linting is mostly routine, sorry about the large files.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            {
+                "default": "array"
+            }
+        ],
+        "@typescript-eslint/ban-types": [
+            "error",
+            {
+                "types": {
+                    "Object": {
+                        "message": "Avoid using the `Object` type. Did you mean `object`?"
```
Comment (jaller94) [inline .eslintrc]: I guess we'd prefer a proper type or even just `Record<string, T|undefined>` or `Record<string, unknown>`.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        "browser": true,
+        "es6": true,
+        "node": true
+    },
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaVersion": 9,
+        "ecmaFeatures": {
+          "jsx": false
+        }
```
Comment (jaller94) [inline .eslintrc]: We don't actually use React here, do we?

```suggestion
```

**Review by jaller94 — APPROVED** [review]
Looks ok.

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,151 @@
+{
+    "env": {
+        "browser": true,
```
Comment (jaller94) [inline .eslintrc]:
```suggestion
```

File: .eslintrc (hunk 2 above)
Comment (Half-Shot) [reply .eslintrc]: Nope, I guess adding a false is overkill :)

**Pull request merged**